api/envoy/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto

```diff
@@ -62,6 +64,88 @@ message UdpProxyConfig {
     }
   }
 
+  // Configuration for tunneling UDP over other transports or application layers.
+  // Tunneling is currently supported over HTTP/2.
+  // [#next-free-field: 10]
+  message UdpTunnelingConfig {
+    // Configuration for UDP datagrams buffering.
+    message BufferOptions {
+      // If set, the filter will only buffer datagrams up to the requested limit, and will drop
+      // new UDP datagrams if the buffer contains the max_buffered_datagrams value at the time
+      // of a new datagram arrival. If not set, the default value is 1024 datagrams.
+      google.protobuf.UInt32Value max_buffered_datagrams = 1;
```

Comment:
It seems like max bytes is always preferable to max datagrams so I'm not sure why someone would use this, but I don't feel strongly about it.

Reply:
The case this was meant to guard was when the downstream sends lots of small UDP datagrams (let's say a lot of 1 byte datagrams). Anyways, this is configurable so if the consumer doesn't need this kind of limit they can set it to max int value